logging/logging.go

```diff
@@ -903,6 +903,15 @@ func deconstructXCloudTraceContext(s string) (traceID, spanID string, traceSampl
 		spanID = ""
 	}
 
+	if spanID != "" {
+		// Convert to 16 byte unsigned hex string
+		intSpanID, err := strconv.ParseUint(spanID, 10, 64)
+		if err != nil || intSpanID == 0 {
+			spanID = ""
+		} else {
+			spanID = fmt.Sprintf("%016x", intSpanID)
+		}
+	}
 	return
```

Comment:
Optional readability suggestion: consider returning the values explicitly rather than relying on named return values. It was a little confusing seeing this return without the list of variables that are returned.